enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,983 @@
+  For more check [this](#Cluster admin wants operator to update the Load Balancer service for updated Ingress Controller eipAllocations)
+  The NLB will get the regular IP address assigned in that case.
+
+8. Why should a custom IngressController should avoid using the EIPs from the cluster config ?
+
+  A custom IngressController cannot use the same EIPs that the default IngressController is using.
+  Usually, the default IngressController is using the EIPs from the cluster config.
+  Therefore, a custom IngressController should not try to use the EIPs from the cluster config
+  because that will fail in the usual case (because the default IngressController is already using them).
+  
+  The default IngressController cannot use the EIPs from the cluster config if a custom
+  IngressController is using them. The expectation is that when the operator creates (or recreates) 
+  the default IngressController, it uses any EIPs that are specified in the cluster config.
+  Therefore, a custom IngressController should avoid using the EIPs from the cluster config
+  so as to avoid preventing the default IngressController from using them.
+
+  If the cluster-admin creates 2 custom IngressControllers, then only one of them could use the EIPs
+  from the cluster config, even if the default IngressController weren't using them.  Using the cluster
+  config in this case would mean the outcome would depend on which custom IngressController were created
+  first. Avoiding using the cluster config for custom IngressControllers makes the behavior more predictable.
+
+9. Are the eips mentioned in the cluster config default for the default Ingress Controller ?  
+   What happens if the cluster-admin deletes the default IngressController and then recreates it before 
+   the polling loop of the operator does, and the cluster-admin doesn't specify eipAllocations?
+
+  No. The eips in the cluster config are only used when the operator creates the default IngressController.
+  If the default Ingress Controller is not found and if the polling loop kicks in and detects it's absence it will
+  create a default IC with the eips mentioned in the cluster config object of ingress.
+  However, it isn't used as a default value when eipAllocations is not specified on the IngressController because
+  the cluster-admin may delete the default IngressController and then recreate it with no eips or no already assigned eips
+  or before the polling loop does.
+  Because the operator only uses the cluster config to specify eipAllocations when the operator itself creates
+  the default IngressController, the cluster-admin can always delete and recreate the default IngressController
+  with eipAllocations deliberately left unspecified, so that the IngressController doesn't use the EIPs from the
+  cluster config. Or the cluster-admin  can always delete and recreate the default IngressController with eipAllocations
+  which are not already specified in the config ingress cluster object.
```

Comment:
Can I update the ingress cluster object EIPAllocations and then delete the `default` IngressController to change the EIPs? I don't think that's mentioned in this EP.

I think you should add another workflow: `Changing the EIPs for the default IngressController` and document either:
1. Changing the ingress `cluster` objects EIPs and deleting the ingress controller
2. Deleting and quickly recreating the `default` IngressController with updated EIPs.

I would vote for #1, as that's a more permanent and less racy solution (i.e. if another cluster admin recreates the `default` IC, then the updated EIPs aren't erased).

Reply:
done